File: frontend/lib/audit-workspace.ts
/** Estate Watch audit workspace — phase navigation and status helpers. */

export const AUDIT_PHASES = [
  "intake",
  "satellite",
  "confidence",
  "risk",
  "sampling",
  "reconciliation",
  "export",
  "attestation",
] as const;

export type AuditPhase = (typeof AUDIT_PHASES)[number];

const UNLOCK_STATUS: Record<AuditPhase, string> = {
  intake: "draft",
  satellite: "intake_complete",
  confidence: "analysis_ready",
  risk: "confidence_mapped",
  sampling: "risk_assessed",
  reconciliation: "sampling_planned",
  export: "field_verified",
  attestation: "export_ready",
};

const STATUS_ORDER = [
  "draft",
  "intake_complete",
  "analysis_ready",
  "confidence_mapped",
  "risk_assessed",
  "sampling_planned",
  "field_verified",
  "export_ready",
  "under_review",
  "attested",
] as const;

function statusIndex(status: string): number {
  const idx = STATUS_ORDER.indexOf(status as (typeof STATUS_ORDER)[number]);
  return idx >= 0 ? idx : 0;
}

export function isAuditPhaseUnlocked(phase: AuditPhase, engagementStatus: string): boolean {
  if (phase === "reconciliation") {
    return statusIndex(engagementStatus) >= statusIndex("sampling_planned");
  }
  return statusIndex(engagementStatus) >= statusIndex(UNLOCK_STATUS[phase]);
}

export function defaultAuditPhase(engagementStatus: string): AuditPhase {
  if (engagementStatus === "draft") return "intake";
  if (engagementStatus === "intake_complete") return "satellite";
  if (engagementStatus === "analysis_ready") return "confidence";
  if (engagementStatus === "confidence_mapped") return "risk";
  if (engagementStatus === "risk_assessed") return "sampling";
  if (engagementStatus === "sampling_planned") return "sampling";
  if (engagementStatus === "field_verified") return "export";
  if (engagementStatus === "export_ready" || engagementStatus === "under_review") return "attestation";
  if (engagementStatus === "attested") return "attestation";
  return "intake";
}

export function formatAuditStatus(status: string): string {
  return status.replaceAll("_", " ");
}

File: mobile/lib/src/audit_workspace.dart
// Helpers for Estate Watch mobile audit workspace.

const auditAttestationEnabledStatuses = {
  'export_ready',
  'under_review',
  'attested',
};

bool auditAttestationEnabled(String? engagementStatus) {
  if (engagementStatus == null || engagementStatus.isEmpty) return false;
  return auditAttestationEnabledStatuses.contains(engagementStatus);
}

String auditEngagementStatusLabel(String status) {
  return switch (status) {
    'draft' => 'Draft',
    'intake_complete' => 'Intake complete',
    'analysis_ready' => 'Analysis ready',
    'confidence_mapped' => 'Confidence mapped',
    'risk_assessed' => 'Risk assessed',
    'sampling_planned' => 'Sampling planned',
    'field_verified' => 'Field verified',
    'export_ready' => 'Export ready',
    'under_review' => 'Under review',
    'attested' => 'Attested',
    'no_engagement' => 'No engagement',
    _ => status.replaceAll('_', ' '),
  };
}

bool auditNeedsFieldPlots(String? engagementStatus) {
  return engagementStatus == 'sampling_planned' || engagementStatus == 'field_verified';
}

bool auditNeedsAttestationAction(Map<String, dynamic>? attestationSummary) {
  if (attestationSummary == null) return false;
  if (attestationSummary['can_sign'] == true) return true;
  if (attestationSummary['can_cosign'] == true) return true;
  final pending = (attestationSummary['review_queue'] as Map?)?['pending_review_count'] as num?;
  return (pending ?? 0) > 0;
}
